src/linear_algebra/eigenspace.lean

```diff
@@ -351,13 +351,29 @@ generalized_eigenspace_mono (nat.succ_le_of_lt hk)
 
 /-- All eigenvalues are generalized eigenvalues. -/
 lemma has_generalized_eigenvalue_of_has_eigenvalue
-  {f : End K V} {μ : K} {k : ℕ} (hk : 0 < k) (hμ : f.has_eigenvalue μ) :
+  {f : End R M} {μ : R} {k : ℕ} (hk : 0 < k) (hμ : f.has_eigenvalue μ) :
   f.has_generalized_eigenvalue μ k :=
 begin
   apply has_generalized_eigenvalue_of_has_generalized_eigenvalue_of_le hk,
   rwa [has_generalized_eigenvalue, generalized_eigenspace, pow_one]
 end
 
+/-- All generalized eigenvalues are eigenvalues. -/
+lemma has_eigenvalue_of_has_generalized_eigenvalue
```

Comment:
`has_generalized_eigenvalue.has_eigenvalue` might be a better name

Reply:
Of course I'm happy to change but I think consistency with the rest of the file might be the greater good here. In particular, this name mirrors `has_generalized_eigenvalue_of_has_eigenvalue`.

Reply:
The advantage would be that dot notation worked, but I agree consistency is good.

Perhaps the right thing to do is leave things as you have them, but make a follow-up PR that converts everything sensible to dotted names, assuming we think that that's desirable to have.